persistence-jpa/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/jpa/Retry.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.javadsl.persistence.jpa
+
+import java.util.concurrent.CompletionStage
+import java.util.function.Supplier
+
+import akka.actor.Scheduler
+import akka.pattern.after
+
+import scala.concurrent.duration.Duration.fromNanos
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.control.NonFatal
+
+// With thanks to https://gist.github.com/viktorklang/9414163
+// TODO: add tests, consider moving into another sub-project
```

Comment:
Leave here for now, when we need it somewhere else it might become evident where ti should live.